include/triton/common/triton_json.h

```diff
@@ -30,6 +30,8 @@
 // RapidJSON's GetObject.
 // https://github.com/Tencent/rapidjson/issues/1448
 #undef GetObject
+// Define NOMINMAX to build successfully with std::min/max
+#define NOMINMAX
```

Comment:
Are we able to define this in cmake so that we don't have to define this in every header file?  I don't believe we use the `MIN`/`MAX` macros from windows at all.

Reply:
I have created a follow-up ticket to explore this options: DLIS-4769.

Reply:
Looks like there is still some std::min/max being included in the library. I am going to experiment with the compiler flags next.

Reply:
Windows.h is probably included earlier (especially given that undef GetObject is called above, which would be called after windows.h per the comments)? You'd need to include it before windows.h is included.

Reply:
@dyastremsky Correct, we don't have the issue until `windows.h` is included. If we are able to define `NOMINMAX` on the compilation line then that define is defined at the start of every header.

Reply:
Thanks for the catch!